itests/hive-unit/src/test/java/org/apache/hive/beeline/TestHplSqlViaBeeLine.java

```diff
@@ -610,12 +612,268 @@ public void testNullCast() throws Throwable {
     testScriptFile(SCRIPT_TEXT, args(), "^(.(?!(NullPointerException)))*$", OutStream.ERR);
   }
 
+  @Test
+  public void testACTIVITY_COUNTHplSqlFunction() throws Throwable {
+    String SCRIPT_TEXT =
+        "DROP TABLE IF EXISTS result;\n" +
+        "CREATE TABLE result (col1 string);\n" +
+        "INSERT INTO result VALUES('Alice');\n" +
+        "INSERT INTO result VALUES('Bob');\n" +
+        "SELECT * FROM result;\n" +
+        "SELECT ACTIVITY_COUNT;";
+    testScriptFile(SCRIPT_TEXT, args(), "2");
+  }
+
+  @Test
+  public void testCASTHplSqlFunction1() throws Throwable {
+    String SCRIPT_TEXT = "SELECT CAST('Abc' AS CHAR(1));";
+    testScriptFile(SCRIPT_TEXT, args(), "A");
+  }
+
+  @Test
+  public void testCASTHplSqlFunction2() throws Throwable {
+    String SCRIPT_TEXT = "SELECT CAST(TIMESTAMP '2015-03-12 10:58:34.111' AS CHAR(10));";
+    testScriptFile(SCRIPT_TEXT, args(), "2015-03-12");
+  }
+
+  @Test
+  public void testCHARHplSqlFunction() throws Throwable {
+    String SCRIPT_TEXT = "select CHAR(2023)";
+    testScriptFile(SCRIPT_TEXT, args(), "2023");
+  }
+
+  @Test
+  public void testCOALESCEHplSQLFunction() throws Throwable {
```

Comment:
I notice this test is not the same as what coalesce.sql was covering.  That test file's contents have been removed, so don't we want those tests to carry over here ?  Are we going to lose any test coverage ?
The same applies to some of the other functions such as SUBSTRING, TRIM, FROM_UNIXTIME etc.

Reply:
COALESCE function returns first non-null value in the input list. coalesce.sql was covering scenarios like no null value, first param null with only second non-null param, first param null with two non-null params and two null params with one non-null param. All these scenarios can be added to TestHplSqlViaBeeLine.java also, will try to add them also.